mos-platform/mega65/crt/divmod.h

```diff
@@ -0,0 +1,244 @@
+#ifndef __SLOW_DIV
+
+#define DIVBUSY 0xD70F
+#define DIVOUT_FRAC 0xD768
+#define DIVOUT_WHOLE 0xD76C
+#define MULTINA 0xD770
+#define MULTINB 0xD774
+#define MULTOUT 0xD778
+
+// Max 32-bit unsigned integer division using MEGA65 accelerated math registers
+template <typename T> static inline T udiv_m65(const T a, const T b) {
+  if (b) {
+    *(volatile unsigned long *)(MULTINA) = a;
+    *(volatile unsigned long *)(MULTINB) = b;
+    while (*(volatile unsigned char *)(DIVBUSY)) {
+    };
+    return *(volatile T *)(DIVOUT_WHOLE);
+  }
+  return 0;
+}
+
+// Relatively straigtforward implementation of long division in C. Not
+// particularly tuned for performance, but clear.
+
+template <typename T> static inline T udiv(T a, T b) {
+  if constexpr (sizeof(T) <= 4) {
+    return udiv_m65(a, b);
+  }
+
+  if (!b || b > a)
```

Comment:
Instead of copying this, just call this udiv_m65, and call the above udiv_m65_hardware or something. Then, this can just call through to the original udiv rather than duplicating it. Same for multiplication.

Reply:
Yes, that's a very good point - I initially removed the original implementation, but discovered that it's needed for 64-bit ints. Will fix.

Reply:
I'm entirely sure what CMake magic to apply to expose the common crt to derived targets. I the header file should be available only during building and not part of the install target.

Reply:
`target_include_directories` set to `PRIVATE` I think; these affect building of the SDK in CMake, but shouldn't affect the installed SDK.